pyproject.toml

```diff
@@ -55,6 +55,16 @@ force-exclude = '''
 '''
 
 
+[tool.docformatter]
+    # The ``summaries`` are not (yet) 75 characters because the summary lines can't be
```

Comment:
The summary lines are problematic. If they are lacking, then it can result in long lines. If they are present they are often longer than 75 characters and must be rewritten.

Reply:
I guess my question was mostly on how to proceed with this practically: the desired end state clearly is a summary line that is short enough, which needs a rewrite. But if the auto-fix mode makes current multi-line summaries into single very long lines, then that would seem worse than keeping it multi-line. I.e., I was wondering if one could enforce some kind of process that never left `main` in an inferior state. E.g., a process where there were 2 commits: one with basic reformatting that could just be glanced over, and another where the summary line rewrites get done (i.e., precommit autofix on the body, failure on the summary).

p.s. Perhaps this brings up a bigger picture: is there a general place where all our precommit stuff is documented? At least, I am not very used yet to check what the auto-stuff has done, and while for code it is almost always fine, for docstrings, I'd be a little more hesitant to trust it (though again, overall the changes here are good).

Reply:
The auto fixes are never pushed without human review, so at least instances of bad docstrings should be noted, as you have done in this PR.
There's two sides to the debate about settings:
1. Multi-line descriptions without a single-line summary will always trip up `docformatter`. With our current setting it turns it into one long line. With the end-goal setting of 75 characters it weirdly breaks apart the multi-line description. Either way it's wrong and needs to be fixed. However, the end-goal setting of 75 characters also necessitates rewriting every summary line that's too long. Both are work. Enforcing 75 characters now is just a lot more work.
2. On the other hand, if we roll this out Astropy-wide then go to change the summary-line setting, everything will start failing!

So 🤷

Reply:
> is there a general place where all our precommit stuff is documented?

Each check fails individually and most have some kind of verbose mode which are useful for diagnostics if something is failing and you're not sure why.
But as for detailed explicit comments? Only what's in in `pre-commit-config.yaml` and `pyproject.toml` itself, I suppose.

Reply:
OK, I see you have little choice with `wrap_summary` - that behaviour does not make too much sense! Anyway, fine to leave it as you have it for now and see how things go. Overall, the changes were quite obviously good!